File: frontend/src/types/agent.ts
export type AgentType = 'scout' | 'tracker' | 'analyst' | 'sleuth' | 'sentinel' | 'orchestrator';

export type AgentStatusValue = 'idle' | 'initializing' | 'working' | 'complete' | 'error';

export interface AgentStatus {
  status: AgentStatusValue;
  progress: number;
  currentTask: string;
}

export interface AgentMessage {
  id: string;
  agentType: AgentType;
  message: string;
  timestamp: number;
  type: 'info' | 'success' | 'warning' | 'error';
}

export const AGENT_CONFIG: Record<AgentType, { displayName: string; color: string; description: string }> = {
  orchestrator: {
    displayName: 'Hive Mind',
    color: 'var(--color-agent-orchestrator)',
    description: 'Coordinates all agents',
  },
  scout: {
    displayName: 'Scout',
    color: 'var(--color-agent-scout)',
    description: 'Fetches on-chain data',
  },
  tracker: {
    displayName: 'Tracker',
    color: 'var(--color-agent-tracker)',
    description: 'Traces money flows',
  },
  analyst: {
    displayName: 'Analyst',
    color: 'var(--color-agent-analyst)',
    description: 'Calculates PNL & scores',
  },
  sleuth: {
    displayName: 'Sleuth',
    color: 'var(--color-agent-sleuth)',
    description: 'Finds wallet connections',
  },
  sentinel: {
    displayName: 'Sentinel',
    color: 'var(--color-agent-sentinel)',
    description: 'Assesses risk levels',
  },
};

File: backend/app/schemas/agent.py
"""Agent-related Pydantic schemas and enumerations."""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel, Field


class AgentType(str, Enum):
    """Enumeration of agent roles in the analysis pipeline."""

    scout = "scout"
    tracker = "tracker"
    analyst = "analyst"
    sleuth = "sleuth"
    sentinel = "sentinel"
    orchestrator = "orchestrator"


class AgentStatusValue(str, Enum):
    """Possible runtime statuses for an agent."""

    idle = "idle"
    initializing = "initializing"
    working = "working"
    complete = "complete"
    error = "error"


class AgentStatusUpdate(BaseModel):
    """Status update emitted by an agent during execution."""

    agent_type: AgentType
    status: AgentStatusValue
    progress: int = Field(ge=0, le=100)
    current_task: str = ""


class AgentMessage(BaseModel):
    """Informational message emitted by an agent."""

    id: str
    agent_type: AgentType
    message: str
    timestamp: float
    type: str = Field(
        default="info",
        description="Message severity: info, success, warning, or error",
    )
